### PR TITLE
[query] Push FS through EmitFunctionBuilder

### PR DESCRIPTION
**Pull request by tpoterba**

### CODE REVIEW

File: hail/src/main/scala/is/hail/expr/ir/EmitClassBuilder.scala [hunk 1]
```diff
@@ -27,42 +27,15 @@ class EmitModuleBuilder(val ctx: ExecuteContext, val modb: ModuleBuilder) {
...
-        val ba = b64.invoke[String, Array[Byte]]("decode", stringAssembler)
-        val bais = Code.newInstance[ByteArrayInputStream, Array[Byte]](ba)
-        val ois = cb.newLocal("ois", Code.newInstance[ObjectInputStream, InputStream](bais))
-        Code.checkcast(ois.invoke[Any]("readObject"))(fsinfo)
-      }
-      val fs = cls.newStaticField[FS]("filesystem", mb.invokeCode())
-
-      _staticFS = new StaticFieldRef(fs)
-    }
-    _staticFS
```
Comment (danking) [inline hail/src/main/scala/is/hail/expr/ir/EmitClassBuilder.scala:63]: love it

**Review by danking — COMMENTED** [review]
I'm good with the changes to service code. Should someone from compilers team do a once over of the rest?

**Review by danking — CHANGES_REQUESTED** [review]
x

**Comment by tpoterba** [comment]
Good call. Assigning to Chris for another look, will move assignee.

**Comment by tpoterba** [comment]
@chrisvittal - the motivation for this change is that we don't want to serialize the FS in the compiled function because it contains secrets. Instead, clients who call compiled functions should pass in the FS.

**Review by chrisvittal — APPROVED** [review]
lgtm

**Pull request merged**